### DISCUSSION TITLE
Deprecating Timm Feature Extractor

### DISCUSSION THREAD

**Discussion by ashwinvaidya17** [open-edge-platform/anomalib#779] [Announcements]
# Deprecating Timm Feature Extractor

We have decided to deprecate Timm in favour of [TorchFX](https://pytorch.org/docs/stable/fx.html).

## Motivation

- Our primary motivation was because we observed huge performance drops when using Timm feature extractor. While this can be attributed to minor differences in architecture and model weights, it is not desirable and hurts our ability to reproduce results.
- We can pass any nn.Module object to it making it more general purpose. This also means that if a user wants to use Timm models, they can just pass the model object to TorchFX and use it.
- It will reduce one dependency for the library.

**Comment by youngquan** [comment]
It seems that we haven't deprecated the `Timm Feature Extractor` yet, and we are currently using an older version, `timm==0.6.13`. What's the plan? Are we intending to continue using `timm`, or do we plan to deprecate it soon? If we are still using `timm`, is it possible to upgrade it to the latest version?
